### PR TITLE
[MNG-8453] Check dependency declarations by default

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

File: maven-plugins/pom.xml [hunk 1]
```diff
@@ -129,6 +129,22 @@ under the License.
...
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>analyze</id>
+            <goals>
+              <goal>analyze-only</goal>
+            </goals>
+            <configuration>
+              <failOnWarning>false</failOnWarning>
```
Comment (slawekjaranowski) [inline maven-plugins/pom.xml:142]: I would like to use a property here - it will be easy to override in child project

Comment (elharo) [reply maven-plugins/pom.xml]: done

File: maven-plugins/pom.xml [hunk 2]
```diff
@@ -129,6 +130,22 @@ under the License.
...
+        <artifactId>maven-dependency-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>analyze</id>
+            <goals>
+              <goal>analyze-only</goal>
+            </goals>
+            <configuration>
+              <failOnWarning>${enforce.dependency.declarations}</failOnWarning>
+              <ignoredDependencies>org.apache.maven:maven-compat:*,org.slf4j:slf4j-simple:*</ignoredDependencies>
```
Comment (slawekjaranowski) [inline maven-plugins/pom.xml:144]: should be as separate item for each ignore artifact to allow append more in child project
`ignoredDependencies` is a list parameter

Comment (slawekjaranowski) [reply maven-plugins/pom.xml:144]: last `*` can also be skipped 

https://maven.apache.org/plugins/maven-dependency-plugin/examples/exclude-dependencies-from-dependency-analysis.html

Comment (elharo) [reply maven-plugins/pom.xml]: good ideas, done

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
for this repo we no need an jira issue ... simply add label and Milestone - release notes will be generate

**Pull request merged**